typescript-generator-core/src/main/java/cz/habarta/typescript/generator/TsType.java

```diff
@@ -16,10 +16,23 @@
     public static final AliasType DateAsNumber = new AliasType("DateAsNumber", "type DateAsNumber = number;");
     public static final AliasType DateAsString = new AliasType("DateAsString", "type DateAsString = string;");
 
+    @Override
+    public boolean equals(Object rhs) {
+        return rhs != null && this.getClass() == rhs.getClass() && this.toString() == rhs.toString();
```

Comment:
You should not use `==` for string comparisons. That was the first test line I suggested to add.